aten/src/ATen/CPUApplyUtils.h

```diff
@@ -235,13 +247,64 @@ apply_op(int64_t numel, int64_t offset, const Op& op, Args... iters) {
   for (int64_t i = 0; i < numel;) {
     for (; iterate_continue(iters...) && i < numel;) {
       op(*iters.data_...);
-      iterate(iters...);
+      iterate(1, iters...);
       i++;
     }
     iterate_overflow(iters...);
   }
 }
 
+inline void apply_kernel(){};
+
+template <typename Op, typename... Args>
+inline void
+apply_kernel(int64_t numel, int64_t offset, const Op& op, Args... iters) {
+  // For 0-dim tensors
+  if (numel == 1 && max_dim(iters...) == 0) {
+    op(1, iters.data_..., iters.strides_[iters.dim_ - 1]...);
```

Comment:
The strides could probably always be `1`, since scalars don't use them anyway

Reply:
True

Reply:
It's not clear how to make it expand to the same number of arguments as iters.data_